### DISCUSSION TITLE
Multiband image processing

### DISCUSSION THREAD

**Discussion by linjjiang** [HALFpipe/HALFpipe#428] [General] [answered]
Hi,

I am wondering if HALFpipe can be used to preprocess multiband imaging data? For example, I don't see any options to import a single-band reference image.

Thank you!
Linjing

**Comment by HippocampusGirl** [comment, marked as answer]
Hi @Linjing95,

HALFpipe can be used for multi-band data. Since preprocessing is done with fMRIPrep under the hood, HALFpipe supports all data types that fMRIPrep can process.

The option to import a single-band reference image is currently not available in the user interface. However, you can manually edit the `spec.json` file in your working directory to provide the information. To do that, simply add an entry to the `files` array inside the file.

Here is an example for how that entry might look:
```json
        {
            "datatype": "func",
            "path": "/fast/work/groups/ag_walter/leadata/hcp/raw/{sub}/unprocessed/3T/tfMRI_{task}_{ses}/{sub}_3T_tfMRI_{task}_{ses}_SBRef.nii.gz",
            "extension": ".nii.gz",
            "suffix": "sbref"
        },
```

After editing, restart HALFpipe, select your working directory and choose `Run without modification`.

**Reply by linjjiang** [reply]
> Hi @Linjing95,
> 
> HALFpipe can be used for multi-band data. Since preprocessing is done with fMRIPrep under the hood, HALFpipe supports all data types that fMRIPrep can process.
> 
> The option to import a single-band reference image is currently not available in the user interface. However, you can manually edit the `spec.json` file in your working directory to provide the information. To do that, simply add an entry to the `files` array inside the file.
> 
> Here is an example for how that entry might look:
> 
> ```json
>         {
>             "datatype": "func",
>             "path": "/fast/work/groups/ag_walter/leadata/hcp/raw/{sub}/unprocessed/3T/tfMRI_{task}_{ses}/{sub}_3T_tfMRI_{task}_{ses}_SBRef.nii.gz",
>             "extension": ".nii.gz",
>             "suffix": "sbref"
>         },
> ```
> 
> After editing, restart HALFpipe, select your working directory and choose `Run without modification`.

Hi @HippocampusGirl,

It worked! Are there any other settings I should change aside from adding an SBRef? For instance:
1) Would you still recommend doing slice-timing correction?
2) Should I choose to remove any confounds (e.g., ICA-AROMA, GS, WM, CSF, motion) from the data (my data is task-based fMRI)?

Thank you! I appreciate it!

**Reply by HippocampusGirl** [reply]
Hi @Linjing95,

If you are combining multiple data sets, then I would use consistent slice-timing options between them. If not, then it's not something that is usually done for fast multi-band data, so I would disable it.

We recommend some settings in the manual at <https://bit.ly/tbfmrimanual>. 
- There seem to be some anecdotal reports that ICA-AROMA does not perform as well with multi-band data, so if you choose to use that I would also add another identical feature without ICA-AROMA for comparison. 
- GS, WM and CSF are not commonly used for task-based data. 
- Motion regressors are generally not needed when doing ICA-AROMA (since it also uses motion data to determine its regressors). It can be a great option when ICA-AROMA is turned off though!

**Reply by linjjiang** [reply]
> Hi @Linjing95,
> 
> If you are combining multiple data sets, then I would use consistent slice-timing options between them. If not, then it's not something that is usually done for fast multi-band data, so I would disable it.
> 
> We recommend some settings in the manual at https://bit.ly/tbfmrimanual.
> 
> * There seem to be some anecdotal reports that ICA-AROMA does not perform as well with multi-band data, so if you choose to use that I would also add another identical feature without ICA-AROMA for comparison.
> * GS, WM and CSF are not commonly used for task-based data.
> * Motion regressors are generally not needed when doing ICA-AROMA (since it also uses motion data to determine its regressors). It can be a great option when ICA-AROMA is turned off though!

This is really helpful! Thanks so much!
